src/features/query-bar/flowTypes.js

```diff
@@ -28,3 +29,5 @@ export type ConditionType = {
     valueKey: string | null,
     valueType: string,
 };
+
+export type ConnectorType = typeof AND | typeof OR;
```

Comment:
Shouldn't this support null? Isn't it a valid value for ConnectorType?

Reply:
Currently in the UX it's not possible to have a null ConnectorType as the default is set to AND.

Reply:
I'm a bit confused. 

Shouldn't ConnectorType exist within ConditionType? How else would the component communicate connector to the V2 endpoint? 

If the answer is yes, then shouldn't the connector be nullable?

Reply:
I'd like to see Acceptance Criteria around this -> how is the Connector communicates to the calling component when user clicks Apply?

Reply:
I don't think ConnectorType necessarily needs to live within ConditionType. Right now since the ConnectorType is either AND | OR across all conditions, we don't need to keep ConnectorType as a property of a Condition object. In the future if we are handling more complex logic then yes we will need ConnectorType to be part of the Condition object but for now since any changes to the current selected Connector is applied across all Conditions, it can remain as a type independent of Condition.

For communicating connector to the V2 endpoint, here is an example of the transform method:
```
transformConditionToSQL = (conditions: Array<Object>) => {
        const { selectedConnector } = this.state;
        let sql = '';
        conditions.forEach((condition, index) => {
            const { valueType } = condition;
            const operators = OPERATORS_FOR_ATTRIBUTE[valueType];
            if (index === 0) {
                sql += 'where ';
                sql += `columnName is ${condition.columnDisplayText} `;
                sql += `and operator is ${operators[0].displayName} `;
                sql += `and value is ${condition.value} `;
            } else {
                sql += `${selectedConnector} `;
                sql += `columnName is ${condition.columnDisplayText} `;
                sql += `and operator is ${operators[0].displayName} `;
                sql += `and value is ${condition.value} `;
            }
        });
        return sql;
    };
```

Reply:
As for a nullable ConnectorType would that only apply to the first Condition? For the 'where' label.

Reply:
The FilterButton and the "transform" logic should be decoupled. 

FilterButton is concerned with _display_ of conditions ("view concerns") and should have no awareness of SQL. It helps with unit testing, avoids bloating FilterButton, and makes it possible to reason about each without having to juggle multiple concepts at one time.

How would you go about decoupling i.e. to communicate Conditions + connector from FilterButton to transform.